docs/ERC721.md

```diff
@@ -0,0 +1,686 @@
+
+Parameters:
+
+```jsx
+operator: felt
+```
+
+Returns:
+
+None.
+
+#### `isApprovedForAll`
+
+Returns if the `operator` is allowed to manage all of the assets of `owner`.
+
+Parameters:
+
+```jsx
+owner: felt
+operator: felt
+```
+
+Returns:
+
+```jsx
+is_approved: felt
+```
+
+---
+
+### IERC721_Metadata API
+```jsx
+func name() -> (name: felt):
+end
+
+func symbol() -> (symbol: felt):
+end
+
+func tokenURI(token_id: Uint256) -> (uri_len: felt, uri: felt*):
+end
+```
+
+#### `name`
+
+Returns the token collection name.
+
+Parameters:
+
+None.
+
+Returns:
+
+```jsx
+name: felt
+```
+
+#### `symbol`
+
+Returns the token collection symbol.
+
+Parameters: 
+
+None.
+
+Returns:
+
+```jsx
+symbol: felt
+```
+
+#### `tokenURI`
+
+Returns the Uniform Resource Identifier (URI) for `token_id` token. If the baseURI is not set, this function will return `[0]`. Otherwise, the returned array will return 3 values: baseURI, token_id.low, token_id.high. This differs from the Solidity implementation where the returned value is a concatenated string of both the baseURI and token ID.
+
+Parameters:
+
+```jsx
+token_id: Uint256
+```
+
+Returns:
+
+```jsx
+uri_len: felt
+uri: felt*
+```
+
+---
+
+### IERC721_Enumerable API
+```jsx
+
+func totalSupply() -> (totalSupply: Uint256):
+end
+
+func tokenByIndex(index: Uint256) -> (token_id: Uint256):
+end
+
+func tokenOfOwnerByIndex(owner: felt, index: Uint256) -> (token_id: Uint256):
+end
+```
+
+#### `totalSupply`
+
+Returns the total amount of tokens stored by the contract.
+
+Parameters: None
+
+Returns:
+
+```jsx
+totalSupply: Uint256
+```
+
+#### `tokenByIndex`
+
+Returns a token ID owned by `owner` at a given `index` of its token list. Use along with [balanceOf](#balanceOf) to enumerate all of `owner`'s tokens.
+
+Parameters:
+
+```jsx
+index: Uint256
+```
+
+Returns:
+
+```jsx
+token_id: Uint256
+```
+
+#### `tokenOfOwnerByIndex`
+
+Returns a token ID at a given `index` of all the tokens stored by the contract. Use along with [totalSupply](#totalSupply) to enumerate all tokens.
+
+Parameters:
+
+```jsx
+owner: felt
+index: Uint256
+```
+
+Returns:
+
+```jsx
+token_id: Uint256
+```
+
+---
+
+### IERC721_Receiver API
+```jsx
+
+func onERC721Received(
+        operator: felt, 
+        _from: felt, 
+        token_id: Uint256, 
+        data_len: felt
+        data: felt*
+    ) -> (selector: felt):
+end
+```
+
+#### `onERC721Received`
+
+Interface for any contract that wants to support safeTransfers from ERC721 asset contracts.
+
+Parameters:
+
+```jsx
+operator: felt
+_from: felt
+token_id: Uint256
+data_len: felt
+data: felt*
+```
+
+Returns:
+
+```jsx
+selector: felt
+```
+---
+
+### IERC165 API
+```jsx
+func supportsInterface(interface_id: felt) -> (success: felt):
+end
+```
+
+#### `supportsInterface`
+
+Returns true if this contract implements the interface defined by `interface_id`.
+
+Parameters:
+
+```jsx
+interface_id: felt
+```
+
+Returns:
+
+```jsx
+success: felt
```

Comment:
maybe this belongs to a 165-specific doc page? Or at least we should separate it enough so it's easy to move in a different PR.